src/connectors/hits/connectHits.ts

```diff
@@ -6,55 +6,62 @@ import {
   addQueryID,
   noop,
 } from '../../lib/utils';
+import {
+  RendererOptions,
+  WidgetFactory,
+  Unmounter,
+  Renderer,
+} from '../../types';
+import { SearchResults } from 'algoliasearch-helper';
 
 const withUsage = createDocumentationMessageGenerator({
   name: 'hits',
   connector: true,
 });
 
-/**
- * @typedef {Object} HitsRenderingOptions
- * @property {Object[]} hits The matched hits from Algolia API.
- * @property {Object} results The complete results response from Algolia API.
- * @property {Object} widgetParams All original widget options forwarded to the `renderFn`.
- */
-
-/**
- * @typedef {Object} CustomHitsWidgetOptions
- * @property {boolean} [escapeHTML = true] Whether to escape HTML tags from `hits[i]._highlightResult`.
- * @property {function(Object[]):Object[]} [transformItems] Function to transform the items passed to the templates.
- */
-
-/**
- * **Hits** connector provides the logic to create custom widgets that will render the results retrieved from Algolia.
- * @type {Connector}
- * @param {function(HitsRenderingOptions, boolean)} renderFn Rendering function for the custom **Hits** widget.
- * @param {function} unmountFn Unmount function called when the widget is disposed.
- * @return {function(CustomHitsWidgetOptions)} Re-usable widget factory for a custom **Hits** widget.
- * @example
- * // custom `renderFn` to render the custom Hits widget
- * function renderFn(HitsRenderingOptions) {
- *   HitsRenderingOptions.widgetParams.containerNode.html(
- *     HitsRenderingOptions.hits.map(function(hit) {
- *       return '<div>' + hit._highlightResult.name.value + '</div>';
- *     })
- *   );
- * }
- *
- * // connect `renderFn` to Hits logic
- * var customHits = instantsearch.connectors.connectHits(renderFn);
- *
- * // mount widget on the page
- * search.addWidgets([
- *   customHits({
- *     containerNode: $('#custom-hits-container'),
- *   })
- * ]);
- */
-export default function connectHits(renderFn, unmountFn = noop) {
+export type HitsRendererOptions<THitsWidgetParams> = {
+  /**
+   * The matched hits from Algolia API.
+   */
+  hits: Array<Record<string, any>>;
+
+  /**
+   * The response from Algolia API.
```

Comment:
I think this is more correct.

```suggestion
   * The response from the Algolia API.
```